test/EncDecTest.cpp

```diff
@@ -1446,6 +1446,166 @@ test_optional()
 	buf.flush();
 }
 
+static void
+test_raw()
+{
+	TEST_INIT(0);
+	using Buf_t = tnt::Buffer<16 * 1024>;
+	Buf_t buf;
+	std::string msg("Hello, test!");
+
+	mpp::encode(buf, 10);
+
+	mpp::encode(buf, msg);
+
+	std::array<int, 3> add_arr = {1, 2, 3};
+	mpp::encode(buf, add_arr);
+
+	std::map<int, int> add_map = {{1, 2}, {3, 4}};
+	mpp::encode(buf, add_map);
+
+	auto run = buf.begin<true>();
+	using it_t = std::remove_reference_t<decltype(run)>;
```

Comment:
Maybe we need a helper for this?

Reply:
What kind of helper?

Reply:
To get the iterator type. AFAIC, you can use `Buf_t::iterator_common<false>` here (or introduce a typedef in `Buffer`)

Reply:
Done.